### PR TITLE
[novnc-core] move novnc to novnc-core, update to newest version, add linting

### LINKED ISSUE #17907: [@types/novnc] implicitly has an 'any' type
- [x] I tried using the `@types/novnc` package and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @smithkl42

```
ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:18 
    TS7006: Parameter 'cfg' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:23 
    TS7006: Parameter 'api' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:28 
    TS7006: Parameter 'defaults' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:38 
    TS7006: Parameter 'v' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:41 
    TS7006: Parameter 'mode' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:47 
    TS7006: Parameter 'type' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:53 
    TS7006: Parameter 'defval' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:42:61 
    TS7006: Parameter 'desc' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:43:19 
    TS7006: Parameter 'cfg' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:43:24 
    TS7006: Parameter 'api' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:43:29 
    TS7006: Parameter 'defaults' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:43:39 
    TS7006: Parameter 'arr' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:90:17 
    TS7006: Parameter 'defaults' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:91:22 
    TS7006: Parameter 'cursor' implicitly has an 'any' type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:117:5 
    TS7010: 'fillRect', which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:118:5 
    TS7010: 'copyImage', which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:119:5 
    TS7010: 'startTile', which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in [at-loader] ./node_modules/@types/novnc/index.d.ts:120:5 
    TS7010: 'subTile', which lacks return-type annotation, implicitly has an 'any' return type.
```

### PR DESCRIPTION
**Pull request by BendingBender**
Fixes  #17907.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/novnc/noVNC/wiki/Modules-API, https://github.com/larryprice/novnc-core
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by BendingBender** [comment]
@smithkl42: Could you please review this PR?

File: types/novnc-core/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    set_onFBUReceive(handler: (rfb: this, fbu: NvFBU) => void): void;
+    get_onFBUComplete(): (rfb: this, fbu: NvFBU) => void;
+    set_onFBUComplete(handler: (rfb: this, fbu: NvFBU) => void): void;
+    get_onFBResize(): (rfb: this, width: number, height: number) => void;
+    set_onFBResize(handler: (rfb: this, width: number, height: number) => void): void;
+    get_onDesktopName(): (rfb: this, name: string) => void;
+    set_onDesktopName(handler: (rfb: this, name: string) => void): void;
+    get_onXvpInit(): (version: number) => void;
+    set_onXvpInit(handler: (rfb: this, name: string) => void): void;
+	get_display(): Display;
```
Comment (sheetalkamat) [inline types/novnc-core/index.d.ts]: Why do you have these get_ and set_ methods? https://github.com/larryprice/novnc-core#usage says the only public methods of this class are: 

> /*
>  An rfb has the following public methods (see lib/rfb.js for more details):
>   * connect(host, port, password, path)
>   * disconnect()
>   * sendPassword(passwd)
>   * sendCtrlAltDel()
>   * xvpOp(ver, op)
>   * xvpShutdown()
>   * xvpReboot()
>   * xvpReset()
>   * sendKey(keysym, code, down)
>   * clipboardPasteFrom(text)
>   * requestDesktopSize(width, height)
> */

Comment (BendingBender) [inline types/novnc-core/index.d.ts]: This is described in their main wiki, which is in a different project. You can find it here: https://github.com/novnc/noVNC/wiki/Modules-API#configuration-attributes

**Comment by mhegazy** [comment]
@andy-ms  `@types/novnc` needs to be deprecated

**Comment by andy-ms** [comment]
:+1:

**Comment by BendingBender** [comment]
Should I do this? This means adding an appropriate entry to notNeededPackages.json, no?

**Comment by andy-ms** [comment]
No, `notNeededPackages.json` is for packages that provide their own typings; for this situation I've just deprecated manually.

**Comment by BendingBender** [comment]
OK, thanks for the clarification.

**Pull request merged**